x/programs/rust/examples/token/src/lib.rs

```diff
@@ -8,6 +8,7 @@ const INITIAL_SUPPLY: i64 = 123456789;
 #[state_keys]
 pub enum StateKeys {
     /// The total supply of the token. Key prefix 0x0.
+    #[state_keys(getter = i64)]
```

Comment:
It's not immediately clear what `getter = i64` means. I think something like `value_type` would make a lot more sense. With that said, I'm not sure that's the approach that we want to take as we would have to annotate every single field. 

There's probably a better option using a declarative macro to define everything. Something like:
```rust
state_schema! {
    MyUnitVariant => i64,
    MyTupleVariant(Address) => String,
}
```
That can construct a `StateKeys` enum with all the state-keys and some other functionality for compile-time guarantees for getting and setting values. 

Before we implement this, I would like to see a proposal for what this looks like for the end-user. Once we agree on something, we can try and tackle an implementation.

Reply:
I don't think that there would be any much better way to avoid annotating every field with the expected return type, for instance this is not an information that we could get in a reliable way.
Renaming the attribute from `getter` to `value_type` would be great, this is a change I will do.
About the `state_schema` declarative macro, I think that my proposition is a bit more idiomatic. This is for instance what the `serde` or `serde_as` proc macros are doing.

Reply:
Also if we want to be able to also set values, we will need to be able to specify keys that should be a getter or setter, or both. In order to stay secure, we should also be able to be able to apply access control, and bound checks on the variable. I think that the setter is much harder to use and could be a pretty bad footgun,

Reply:
Also, this is something that *may* be unrelated, but the token example features a `Balance(Address)` state key which does not have the default implementation. If the value is not present for an address, then it won't panic but return the default value. Should we allow the users of this macro to specify if they want to have a default value ? It seems that we are starting to get into macros that get in the way of the users because they are way too complex for what they may do.